leap_ec/multiobjective/nsga2.py

```diff
@@ -28,13 +29,18 @@ def nsga_2(max_generations: int, pop_size: int, problem, representation,
           "A Fast and Elitist Multiobjective Genetic Algorithm: NSGA-II." IEEE
           transactions on evolutionary computation 6, no. 2 (2002): 182-197.
 
+        - Bogdan Burlacu. 2022. Rank-based Non-dominated Sorting. arXiv.
+          DOI:https://doi.org/10.48550/ARXIV.2203.13654 
+
```

Comment:
It would be nice to have a doctest.  An example of what this looks like is in the docstring for `generational_ea` [here](https://github.com/AureumChaos/LEAP/blob/dba570f6c7153e5dc43e126815030612aee9058a/leap_ec/algorithm.py#L78).

Doctests for high-level algorithms like this are verbose, but nice because they mean a complete usage example is included in the docs.

Reply:
I wrote one, its fairly rudimentary but it serves as a usage example. I used `random_selection` rather than `tournament_selection` in the example to match the original paper.